app/assets/javascripts/shared/SharedService.js.coffee

```diff
@@ -18,6 +18,40 @@ SharedService = ($http, $state, $window, $document) ->
     '@',
     '(?:[a-zA-Z0-9](?:[a-zA-Z0-9-]*[a-zA-Z0-9])?\\.)+[a-zA-Z0-9](?:[a-zA-Z0-9-]*[a-zA-Z0-9])?'
   ].join(''))
+  Service.languageMap =
+    en: 'English'
+    es: 'Spanish'
+    tl: 'Filipino'
+    zh: 'Chinese'
+
+  Service.getLanguageCode = (langName) ->
+    # will take "English" and return "en", for example
+    if langName
+      capitalizedLangName = langName.charAt(0).toUpperCase() + langName.slice(1)
```

Comment:
`_.startCase` also works for this.

Reply:
🙏 yes thank you i was casting about for something like that, much preferable